server/src/main/java/org/opensearch/search/aggregations/AggregatorFactories.java

```diff
@@ -235,7 +235,7 @@ private static AggregatorFactories.Builder parseAggregators(XContentParser parse
             }
         }
 
-        return factories;
+        return factories.count() > 0 ? factories : null;
```

Comment:
This value is checked for null to determine if there are any aggregations present or not.
https://github.com/opensearch-project/OpenSearch/blob/8eea7b986453271cd227a7b98ee09e70d5e28634/server/src/main/java/org/opensearch/action/search/QueryPhaseResultConsumer.java#L117